### PR TITLE
Made the required fields mandatory in respective forms

### LINKED ISSUE #2557: Required Fields can be submitted empty in some forms
## When updating the shift request form we can submit the form being the "reason for shift" field empty which is a required field.

https://user-images.githubusercontent.com/59426397/170803421-08803e06-4fe1-47aa-aa44-d1f5dee4383b.mp4

## When updating the facility details form we can submit the form being the "address" field empty. The form does not post the details but it should prevent from submitting the form by displaying an error msg that the field is empty.

https://user-images.githubusercontent.com/59426397/170803591-3c0c7061-7236-4f38-ac2b-e37cc0e7f13a.mp4

## When updating the resource request form we can submit the form being the "Description for Request" field empty. 

https://user-images.githubusercontent.com/59426397/170803698-1085718e-effc-4e37-a603-23e81be756bc.mp4

### PR DESCRIPTION
**Pull request by patelaryan7751**
Fixes #2557 

# Bugs Fixed

## Made required fields mandatory in shifting request update form
https://user-images.githubusercontent.com/59426397/174465083-6287cce9-bcd5-4325-b5d5-0684a24a9384.mp4

## Made the address field not accept space only 1 space as an input in the facility update form
https://user-images.githubusercontent.com/59426397/174465114-cfff3378-c7d3-4f9e-8a87-bda6c2b69d51.mp4

## Made the required fields mandatory in resource request update form
https://user-images.githubusercontent.com/59426397/174465275-38a13102-bf8c-4751-8ca9-d0f7a41f0906.mp4

### CODE REVIEW

**Comment by aeswibon** [comment]
@patelaryan7751 You have to reset Gigin's recent commit in netlify.toml. Gigin recently updated netlify.toml to use NPM.

**Comment by patelaryan7751** [comment]
Deploy preview ready https://deploy-preview-2768--care-net.netlify.app/

**Review by cp-Coder — APPROVED** [review]

**Review by cp-Coder — APPROVED** [review]

**Comment by nihal467** [comment]
test approved

File: src/Components/Shifting/ShiftDetailsUpdate.tsx [hunk 1]
```diff
@@ -166,7 +171,7 @@ export const ShiftDetailsUpdate = (props: patientShiftProps) => {
   const handleSubmit = async (e: any) => {
     const validForm = validateForm();
 
-    if (validForm) {
+    if (!validForm) {
```
Comment (khavinshankar) [inline src/Components/Shifting/ShiftDetailsUpdate.tsx:174]: @patelaryan7751 can you please double check if it should be `!validForm`

File: src/Components/Resource/ResourceDetailsUpdate.tsx [hunk 2]
```diff
@@ -159,7 +167,7 @@ export const ResourceDetailsUpdate = (props: resourceProps) => {
   const handleSubmit = async (e: any) => {
     const validForm = validateForm();
 
-    if (validForm) {
+    if (!validForm) {
```
Comment (khavinshankar) [inline src/Components/Resource/ResourceDetailsUpdate.tsx:170]: @patelaryan7751 can you please double check if it should be `!validForm`

**Review by khavinshankar — CHANGES_REQUESTED** [review]
@patelaryan7751 can you please verify and confirm?

File: src/Components/Shifting/ShiftDetailsUpdate.tsx (hunk 1 above)
Comment (patelaryan7751) [reply src/Components/Shifting/ShiftDetailsUpdate.tsx:174]: Yes, before it was not working as expected. So I changed it to make it work as expected.

File: src/Components/Resource/ResourceDetailsUpdate.tsx (hunk 2 above)
Comment (patelaryan7751) [reply src/Components/Resource/ResourceDetailsUpdate.tsx:170]: Yes, before it was not working as expected. So I changed it to make it work as expected.

Comment (khavinshankar) [inline src/Components/Resource/ResourceDetailsUpdate.tsx:170]: @patelaryan7751 went through the code, it seems we are returning weather the form is invalid or not, the variable names are not correct and confusing, so can you please change the variable name to invalidForm or can you negate the return in validateForm.

I think second option would be better, do change which works for you.

File: src/Components/Shifting/ShiftDetailsUpdate.tsx (hunk 1 above)
Comment (khavinshankar) [inline src/Components/Shifting/ShiftDetailsUpdate.tsx:174]: do the same for this also

**Review by khavinshankar — CHANGES_REQUESTED** [review]

**Comment by khavinshankar** [comment]
@patelaryan7751 can you please work on the requested changes and resolve the merge conflict

**Comment by patelaryan7751** [comment]
@khavinshankar yeah sure 👍

**Pull request closed without merging**